partner_event/wizard/res_partner_register_event.py

```diff
@@ -26,6 +28,25 @@ def _prepare_registration(self, partner):
     def button_register(self):
         registration_obj = self.env['event.registration']
         partner_obj = self.env['res.partner']
+        errors = []
         for partner_id in self.env.context.get('active_ids', []):
             partner = partner_obj.browse(partner_id)
-            registration_obj.create(self._prepare_registration(partner))
+            try:
+                with self.env.cr.savepoint():
+                    registration_obj.create(
+                        self._prepare_registration(partner))
+            except:
+                errors.append(partner.name)
+        if errors:
+            self.errors = '\n'.join(errors)
+            data_obj = self.env.ref('partner_event.'
+                                    'res_partner_register_event_view')
+            return {
+                'type': 'ir.actions.act_window',
+                'res_model': self._name,
+                'view_mode': 'form',
+                'view_type': 'form',
+                'view_id': [data_obj.id],
+                'res_id': self.id,
+                'target': 'new',
+            }
```

Comment:
It's closed independently of what you return if I don't remember bad, but we can try.